core/documents-manager.php

```diff
@@ -275,7 +275,10 @@ public function get_doc_or_auto_save( $id, $user_id = 0 ) {
 	 * @return false|Document The document if it exist, False otherwise.
 	 */
 	public function get_doc_for_frontend( $post_id ) {
-		if ( is_preview() || Plugin::$instance->preview->is_preview_mode() ) {
+		$is_preview = is_preview() && isset( $_GET['preview_id'] ) && $post_id === (int) Utils::get_super_global_value( $_GET, 'preview_id' );
```

Comment:
Do you need the isset( $_GET['preview_id'] )? IIRC Utils::get_super_global_value() already handles the isset.